src/test/java/uk/gov/pay/directdebit/mandate/services/MandateServiceTest.java

```diff
@@ -221,6 +224,38 @@ public void confirm_shouldNotConfirmOnDemandMandateForInvalidState() {
         service.confirm(gatewayAccount, mandate, mandateConfirmationRequest);
     }
 
+    @Test
+    public void shouldThrowUnlinkedGCAccountException_onMandateCreationWithUnlinkedAccount() {
+        final String EXTERNAL_ID = "external1d";
+        final String DESCRIPTION = "is awesome";
+        GatewayAccount gatewayAccount = aGatewayAccountFixture()
+                .withExternalId(EXTERNAL_ID)
+                .withDescription(DESCRIPTION)
+                .withPaymentProvider(PaymentProvider.GOCARDLESS)
+                .withType(GatewayAccount.Type.TEST)
+                .withAccessToken(null)
+                .toEntity();
+        when(gatewayAccountDao.findByExternalId(gatewayAccount.getExternalId())).thenReturn(Optional.of(gatewayAccount));
+        boolean exceptionThrown = false;
+        try {
+            service.createMandate(null, gatewayAccount.getExternalId());
+        } catch (UnlinkedGCMerchantAccountException e) {
+            exceptionThrown = true;
+        }
+        assertThat(exceptionThrown, is(true));
```

Comment:
Does the job, but we tend to use 
```
@Rule
public ExpectedException thrown = ExpectedException.none();
```
at the top of the class, then in the test we can do
```
thrown.expect(UnlinkedGcMerchantAccountException.class);
```

Reply:
Fixed both tests :)